pkg/actors/targets/workflow/orchestrator/activity.go

```diff
@@ -67,15 +67,29 @@ func (o *orchestrator) callActivity(ctx context.Context, e *backend.HistoryEvent
 		return err
 	}
 
-	targetActorID := buildActivityActorID(o.actorID, e.GetEventId(), generation)
+	activityActorType := o.activityActorType
+	var targetAppID, sourceAppID string
+
+	if router := e.GetRouter(); router != nil {
+		targetAppID = router.GetTarget()
+		sourceAppID = router.GetSource()
+		log.Debugf("Cross-app activity call: target appID=%s, source appID=%s", targetAppID, sourceAppID)
```

Comment:
We don't need to be logging this- workflow is already incredibly chatty, and workflow instances are logged with their types already.
```suggestion
```